reference/test/base/batch_multi_vector_kernels.cpp

```diff
@@ -0,0 +1,425 @@
+              {{I<T>({1.0, -1.0, 1.5}), I<T>({-2.0, 2.0, 3.0})},
+               {{1.0, -2.0, -0.5}, {1.0, -2.5, 4.0}}},
+              exec)),
+          mtx_00(gko::initialize<DenseMtx>(
+              {I<T>({1.0, -1.0, 1.5}), I<T>({-2.0, 2.0, 3.0})}, exec)),
+          mtx_01(gko::initialize<DenseMtx>(
+              {I<T>({1.0, -2.0, -0.5}), I<T>({1.0, -2.5, 4.0})}, exec)),
+          mtx_1(
+              gko::batch_initialize<Mtx>({{{1.0, -1.0, 2.2}, {-2.0, 2.0, -0.5}},
+                                          {{1.0, 2.5, 3.0}, {1.0, 2.0, 3.0}}},
+                                         exec)),
+          mtx_10(gko::initialize<DenseMtx>(
+              {I<T>({1.0, -1.0, 2.2}), I<T>({-2.0, 2.0, -0.5})}, exec)),
+          mtx_11(gko::initialize<DenseMtx>({{1.0, 2.5, 3.0}, {1.0, 2.0, 3.0}},
+                                           exec)),
+          mtx_2(gko::batch_initialize<Mtx>(
+              {{{1.0, 1.5}, {6.0, 1.0}, {-0.25, 1.0}},
+               {I<T>({2.0, -2.0}), I<T>({1.0, 3.0}), I<T>({4.0, 3.0})}},
+              exec)),
+          mtx_20(gko::initialize<DenseMtx>(
+              {I<T>({1.0, 1.5}), I<T>({6.0, 1.0}), I<T>({-0.25, 1.0})}, exec)),
+          mtx_21(gko::initialize<DenseMtx>(
+              {I<T>({2.0, -2.0}), I<T>({1.0, 3.0}), I<T>({4.0, 3.0})}, exec)),
+          mtx_3(gko::batch_initialize<Mtx>(
+              {{I<T>({1.0, 1.5}), I<T>({6.0, 1.0})}, {{2.0, -2.0}, {1.0, 3.0}}},
+              exec)),
+          mtx_30(gko::initialize<DenseMtx>({I<T>({1.0, 1.5}), I<T>({6.0, 1.0})},
+                                           exec)),
+          mtx_31(gko::initialize<DenseMtx>(
+              {I<T>({2.0, -2.0}), I<T>({1.0, 3.0})}, exec)),
+          mtx_4(gko::batch_initialize<Mtx>(
+              {{{1.0, 1.5, 3.0}, {6.0, 1.0, 5.0}, {6.0, 1.0, 5.5}},
+               {{2.0, -2.0, 1.5}, {4.0, 3.0, 2.2}, {-1.25, 3.0, 0.5}}},
+              exec)),
+          mtx_5(gko::batch_initialize<Mtx>(
+              {{{1.0, 1.5}, {6.0, 1.0}, {7.0, -4.5}},
+               {I<T>({2.0, -2.0}), I<T>({1.0, 3.0}), I<T>({4.0, 3.0})}},
+              exec)),
+          mtx_6(gko::batch_initialize<Mtx>(
+              {{{1.0, 0.0, 3.0}, {0.0, 3.0, 0.0}, {0.0, 1.0, 5.0}},
+               {{2.0, 0.0, 5.0}, {0.0, 1.0, 0.0}, {0.0, -1.0, 8.0}}},
+              exec))
+    {}
+
+    std::shared_ptr<const gko::ReferenceExecutor> exec;
+    std::unique_ptr<Mtx> mtx_0;
+    std::unique_ptr<DenseMtx> mtx_00;
+    std::unique_ptr<DenseMtx> mtx_01;
+    std::unique_ptr<Mtx> mtx_1;
+    std::unique_ptr<DenseMtx> mtx_10;
+    std::unique_ptr<DenseMtx> mtx_11;
+    std::unique_ptr<Mtx> mtx_2;
+    std::unique_ptr<DenseMtx> mtx_20;
+    std::unique_ptr<DenseMtx> mtx_21;
+    std::unique_ptr<Mtx> mtx_3;
+    std::unique_ptr<DenseMtx> mtx_30;
+    std::unique_ptr<DenseMtx> mtx_31;
+    std::unique_ptr<Mtx> mtx_4;
+    std::unique_ptr<Mtx> mtx_5;
+    std::unique_ptr<Mtx> mtx_6;
+
+    std::ranlux48 rand_engine;
```

Comment:
change the ranlux48 to the default one to avoid SFIANE issue in Windows https://github.com/ginkgo-project/ginkgo/pull/954